source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+
+The issue is that the render of an .html file should always return an OutputBuffer, since by definition the content is HTML safe. Regardless of the context, be it render template, partial, or whatever, it’s because of that that [we introduced a new HTML handler](https://github.com/rails/rails/commit/8a998b0fa7523c2c8eb6d0cf56e40408bf6e9b2e).
+
+With this commit, after a `bundle update rails`, the footer appears in the correct way and the header “improves”, but now it shows the ERB code on the page instead of interpreting it. The problem is that the header contains ERB code but the name of the file is  `_header.html`. In Rails 4.2.5, this worked well because by default the ERB Handler processed it. In Rails 5.0, prior to our change, the Raw handler processed it and now the HTML handler processes it because the file ends in .html and the latter is the correct behavior. In our case, the incorrect thing is that the name of the file is `_header.html` and it contains ERB code. The name should have always been `_header.html.erb`. We rename the file and the header begins to be rendered in the correct manner.
+
+And with this, we have passing tests and a properly functioning application.
+
+
+## To wrap up, we synchronize our files the Rails 5 way
+
+We have passing tests and the application is functioning. This is a good time to polish some of the details, fix warnings and adapt our code a bit more to the new version of Rails.
+
+For this, one of the first things that can be done is to run `bin/rails rails:update`. The process will ask us file by file if we want to overwrite them with the new versions. We can see the diff at that moment and, after this, decide or we can make the decision to modify them by hand afterwards. I prefer to start up a new application in another directory, generate a scaffold and go file by file comparing and choosing the new one and/or the one that interests me.
+
+Like usual, a few initializers are added, there are a few changes in config/environments/, an app/models/application_record.rb is added and the models now inherit from ApplicationRecord. We can influence these changes.
+
+Within the new initializers, `config/initializers/active_record_belongs_to_required_by_default.rb` is added, which makes it so that the belongs_to cannot be nil. In our case, we need some to be nil, so we have to add the option `optional: true` to the relations. For example, `belongs_to :category, optional: true`.
+
+Also, config/initializers/callback_terminator.rb is added, which makes it so that the callbacks do not halt more when a false is returned. The way to halt now is with `throw(:abort)`. In our application, this was irrelevant.
```

Comment:
`this was irrelevant` => `this was irrelevant and had no effect` or just `had no effect`